lib/spi/clone/CachingDirectoryManager.ts

```diff
@@ -36,38 +37,37 @@ const cache = new StableDirectoryManager({
  */
 export const CachingDirectoryManager: DirectoryManager = {
 
-    directoryFor(owner: string, repo: string, branch: string, opts: CloneOptions): Promise<CloneDirectoryInfo> {
-
-        return cache.directoryFor(owner, repo, branch, opts).then(existing =>
-            pleaseLock(existing.path).then(lockResult => {
-                if (lockResult.success) {
-                    incrementReuse(owner, repo);
-                    return {
-                        ...existing,
-                        release: () => {
-                            logger.debug("Releasing lock on '%s'", existing.path);
-                            return lockResult.release().then(existing.release);
-                        },
-                        invalidate: () => {
-                            logger.debug("Invalidating '%s'", existing.path);
-                            return cache.invalidate(existing)
-                                .then(() => {
-                                    logger.debug("Invalidated. Now releasing lock");
-                                    return lockResult.release().then(existing.release);
-                                });
-                        },
-                        provenance: (existing.provenance || "") + " successfully locked",
-                    };
-                } else {
-                    logger.debug("Lock detected on '%s'", existing.path);
-                    incrementFallback(owner, repo);
-                    return TmpDirectoryManager.directoryFor(owner, repo, branch, opts).then(cdi =>
-                        ({
-                            ...cdi,
-                            provenance: `Tried '${existing.path}' but it was locked. ` + (cdi.provenance || ""),
-                        }));
-                }
-            }));
+    async directoryFor(owner: string, repo: string, branch: string, opts: CloneOptions): Promise<CloneDirectoryInfo> {
+        const existing = await cache.directoryFor(owner, repo, branch, opts);
+        try {
+            const lockResult = await lockfile.lock(existing.path);
+            incrementReuse(owner, repo);
+            return {
+                ...existing,
+                release: () => {
+                    logger.debug("Releasing lock on '%s'", existing.path);
+                    return lockResult().then(existing.release);
+                },
+                invalidate: () => {
+                    logger.debug("Invalidating '%s'", existing.path);
+                    return cache.invalidate(existing)
+                        .then(() => {
+                            logger.debug("Invalidated. Now releasing lock");
+                            return lockResult().then(existing.release);
```

Comment:
I'm going to investigate `existing.release` further.  I"m not sure how this is actually consumed.